File: leetcode/problems/problem-specific/26.remove-duplicates-from-sorted-array/26.remove-duplicates-from-sorted-array.go
package main

func removeDuplicates(nums []int) int {
	if len(nums) <= 1 {
		return len(nums)
	}

	i := 0
	for j := 1; j < len(nums); j += 1 {
		if nums[i] != nums[j] {
			i += 1
			nums[i] = nums[j]
		}
	}

	return i + 1
}

File: leetcode/problems/problem-specific/26.remove-duplicates-from-sorted-array/26.remove-duplicates-from-sorted-array.py
#
# @lc app=leetcode id=26 lang=python3
#
# [26] Remove Duplicates from Sorted Array
#

# @lc code=start

# * Two Pointers Solution | O(n) Time | O(1) Space
# * n -> The length of nums array


class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        left = 0
        for right in range(1, len(nums)):
            if nums[left] != nums[right]:
                left += 1
                nums[left] = nums[right]

        return left + 1


# @lc code=end
